server/src/main/java/org/apache/iotdb/db/service/metrics/SystemMetrics.java

```diff
@@ -51,21 +51,25 @@ public class SystemMetrics implements IMetricSet {
   private Future<?> currentServiceFuture;
   private final ScheduledExecutorService service = Executors.newSingleThreadScheduledExecutor();
   private final Set<FileStore> fileStores = new HashSet<>();
+  private boolean isDataNode = false;
   private long systemDiskTotalSpace = 0L;
   private long systemDiskFreeSpace = 0L;
 
-  public SystemMetrics() {
-    osMXBean = (OperatingSystemMXBean) ManagementFactory.getOperatingSystemMXBean();
+  public SystemMetrics(boolean isDataNode) {
+    this.isDataNode = isDataNode;
+    this.osMXBean = (OperatingSystemMXBean) ManagementFactory.getOperatingSystemMXBean();
   }
 
   @Override
   public void bindTo(AbstractMetricService metricService) {
     collectSystemCpuInfo(metricService);
-    collectSystemDiskInfo(metricService);
+    if (isDataNode) {
+      collectSystemDiskInfo(metricService);
+    }
     collectSystemMemInfo(metricService);
 
     // finally start to update the value of some metrics in async way
-    if (metricService.isEnable() && null == currentServiceFuture) {
+    if (metricService.isEnable() && null == currentServiceFuture && isDataNode) {
```

Comment:
Let's combine the logic into collectSystemDiskInfo because it seems that this scheduled task is only serviced for `collectSystemDiskInfo `

Reply:
Already Fixed~